### PR TITLE
add substations to elasticsearch indexation

### PR DESCRIPTION
**Pull request by achour94**

### CODE REVIEW

**Review by FranckLecuyer — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
Need a test

**Review by FranckLecuyer — APPROVED** [review]

File: src/test/java/org/gridsuite/modification/server/service/BuildTest.java [hunk 1]
```diff
@@ -349,6 +349,125 @@ public void runBuildWithEmptyGroupTest() throws Exception {
...
+                .voltageLevels(Set.of(VoltageLevelInfos.builder().id("v1").name("v1").build(), VoltageLevelInfos.builder().id("v2").name("v2").build()))
+                .substations(Set.of(SubstationInfos.builder().id("s1").name("s1").build()))
+                .build();
+        equipmentInfos = equipmentInfosRepository.findByIdInAndNetworkUuidAndVariantId(List.of("newLine"), TEST_NETWORK_ID, NetworkCreation.VARIANT_ID).get(0);
+        assertTrue(areEquipmentInfosEqual(expectedEquipmentInfos, equipmentInfos));
+
+        assertTrue(TestUtils.getRequestsDone(1, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
+    }
+
+    public static boolean areEquipmentInfosEqual(EquipmentInfos equipmentInfos1, EquipmentInfos equipmentInfos2) {
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/service/BuildTest.java:461]: You can use :       assertThat(equipmentInfos1).recursivelyEquals(equipmentInfos2);

**Review by FranckLecuyer — COMMENTED** [review]
Use assertJ to do the comparaison

File: src/test/java/org/gridsuite/modification/server/service/BuildTest.java (hunk 1 above)
Comment (achour94) [reply src/test/java/org/gridsuite/modification/server/service/BuildTest.java:459]: i created a method to exclude unique_id variable from the test i want to check only other variables indexed in ES

File: src/test/java/org/gridsuite/modification/server/service/BuildTest.java [hunk 2]
```diff
@@ -349,6 +349,123 @@ public void runBuildWithEmptyGroupTest() throws Exception {
...
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/service/BuildTest.java:459]: Remove this method and use an utility method for  equality test
Or use directly the `equals` method of `EquipmentInfo`

File: src/test/java/org/gridsuite/modification/server/service/BuildTest.java [hunk 3]
```diff
@@ -349,6 +350,122 @@ public void runBuildWithEmptyGroupTest() throws Exception {
...
+                .networkUuid(TEST_NETWORK_ID)
+                .variantId(NetworkCreation.VARIANT_ID)
+                .id("vl1")
+                .name("vl1")
+                .type(IdentifiableType.VOLTAGE_LEVEL.name())
+                .voltageLevels(Set.of(VoltageLevelInfos.builder().id("vl1").name("vl1").build()))
+                .substations(Set.of(SubstationInfos.builder().id("s1").name("s1").build()))
+                .build();
+        var equipmentInfos = equipmentInfosRepository.findByIdInAndNetworkUuidAndVariantId(List.of("vl1"), TEST_NETWORK_ID, NetworkCreation.VARIANT_ID).get(0);
+        Assertions.assertThat(equipmentInfos)
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/service/BuildTest.java:429]: Remove `Assertions` and import `assertThat` ?

**Review by SlimaneAmar — APPROVED** [review]

File: src/test/java/org/gridsuite/modification/server/service/BuildTest.java (hunk 3 above)
Comment (achour94) [reply src/test/java/org/gridsuite/modification/server/service/BuildTest.java:429]: we already have another assertThat from org.hamcrest.MatcherAssert.assertThat

**Pull request merged**